src/state/orders/reducer.ts

```diff
@@ -45,27 +48,38 @@ export default createReducer<OrderState>(initialState, (builder) =>
         orders,
       }
     })
+    .addCase(cancelOrders, (state: OrderState, { payload: { orderId } }) => {
+      const newOrders = [...new Set(state.orders)].map((order) =>
+        orderId === order.id ? { ...order, status: OrderStatus.PENDING_CANCELLATION } : order,
+      )
+      return {
+        ...state,
+        orders: newOrders,
+      }
+    })
     .addCase(removeOrders, (state: OrderState, { payload: { orderId } }) => {
       const newOrders = [...new Set(state.orders)].filter((order) => !(orderId === order.id))
       return {
         ...state,
         orders: newOrders,
       }
     })
-    .addCase(finalizeOrderPlacement, (state: OrderState) => {
-      const orders = []
-      for (const order of state.orders) {
-        orders.push({
-          id: order.id,
-          sellAmount: order.sellAmount,
-          price: order.price,
-          status: OrderStatus.PLACED,
-          chainId: order.chainId,
-        })
+    .addCase(finalizeOrderCancellation, (state: OrderState) => {
+      const newOrders = [...new Set(state.orders)].filter(
```

Comment:
What happens if we have two pending cancellations and only 1 really finalized?

We do the same simplification for `finalizeOrderPlacement` 😊 .

I think its not so important for now, as it would be an exceptional case. Maybe you can leave a comment.